packages/langium/src/generator/generator-node.ts

```diff
@@ -59,6 +68,8 @@ export class CompositeGeneratorNode {
 
     readonly contents: Array<(GeneratorNode | string)> = [];
 
+    tracedSource?: TraceSourceSpec;
```

Comment:
Suggestion: We should make this readonly as well:
```suggestion
    get tracedSource(): TraceSourceSpec | undefined {
        return this._tracedSource;
    }
    
    private _tracedSource?: TraceSourceSpec;
```

Reply:
I don't think so, as the content of `contents` can also be manipulated externally.
Besides the current `trace` method doesn't allow to reset this field, so I wouldn't be pedantic at this point.

If somebody wants to break something: feel free!